{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Using manual alignments \n",
    "\n",
    "This recipe shows how run a phylogenetic workflow ignoring the sequence alignment phase. This allows to treat the input sequences as _aligned sequences_, and being used as it in next steps. \n",
    "\n",
    "\n",
    "## Requirements\n",
    "- ete3\n",
    "- ete3_external_apps\n",
    "- [basic concepts about ete-build](ete_build_basics.ipynb) \n",
    "- [composing custom workflows](ete_build_workflows.ipynb)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Recipe \n",
    "\n",
    "The only required step to use manual alignments is to set to *`none`* the alignining task in your preferred workflow name. \n",
    "\n",
    "For instance, the following run, will run modeltesting and a raxml tree over a pre-aligned set of sequences: \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Toolchain path: /Users/jhc/anaconda/bin/ete3_apps \n",
      "Toolchain version: 2.0.3\n",
      "\n",
      "\n",
      "      --------------------------------------------------------------------------------\n",
      "                  ETE build - reproducible phylogenetic workflows \n",
      "                                    unknown, unknown.\n",
      "\n",
      "      If you use ETE in a published work, please cite:\n",
      "\n",
      "        Jaime Huerta-Cepas, Joaquín Dopazo and Toni Gabaldón. ETE: a python\n",
      "        Environment for Tree Exploration. BMC Bioinformatics 2010,\n",
      "        11:24. doi:10.1186/1471-2105-11-24\n",
      "\n",
      "      (Note that a list of the external programs used to complete all necessary\n",
      "      computations will be also shown after execution. Those programs should\n",
      "      also be cited.)\n",
      "      --------------------------------------------------------------------------------\n",
      "\n",
      "    \n",
      "\u001b[32mINFO\u001b[0m -  Testing x86-64  portable applications...\n",
      "       clustalo: \u001b[32mOK\u001b[0m - 1.2.1\n",
      "\u001b[33mDialign-tx not supported in OS X\u001b[0m\n",
      "       fasttree: \u001b[32mOK\u001b[0m - FastTree Version 2.1.8 Double precision (No SSE3), OpenMP (1 threads)\n",
      "         kalign: \u001b[32mOK\u001b[0m - Kalign version 2.04, Copyright (C) 2004, 2005, 2006 Timo Lassmann\n",
      "          mafft: \u001b[32mOK\u001b[0m - MAFFT v6.861b (2011/09/24)\n",
      "         muscle: \u001b[32mOK\u001b[0m - MUSCLE v3.8.31 by Robert C. Edgar\n",
      "          phyml: \u001b[32mOK\u001b[0m - . This is PhyML version 20160115.\n",
      "     pmodeltest: \u001b[32mOK\u001b[0m - pmodeltest.py v1.4\n",
      "          prank: \u001b[32mOK\u001b[0m - prank v.100802. Minimal usage: 'prank sequence_file'\n",
      "       probcons: \u001b[32mOK\u001b[0m - PROBCONS version 1.12 - align multiple protein sequences and print to standard output\n",
      "          raxml: \u001b[32mOK\u001b[0m - This is RAxML version 8.1.20 released by Alexandros Stamatakis on April 18 2015.\n",
      " raxml-pthreads: \u001b[32mOK\u001b[0m - This is RAxML version 8.1.20 released by Alexandros Stamatakis on April 18 2015.\n",
      "         readal: \u001b[32mOK\u001b[0m - readAl v1.4.rev6 build[2012-02-02]\n",
      "         statal: \u001b[32mOK\u001b[0m - statAl v1.4.rev6 build[2012-02-02]\n",
      "        tcoffee: \u001b[32mOK\u001b[0m - PROGRAM: T-COFFEE Version_11.00.8cbe486 (2014-08-12 22:05:29 - Revision 8cbe486 - Build 477)\n",
      "         trimal: \u001b[32mOK\u001b[0m - trimAl v1.4.rev6 build[2012-02-02]\n",
      "\u001b[32mINFO\u001b[0m -  Starting ETE-build execution at Mon Feb  8 14:17:42 2016\n",
      "\u001b[32mINFO\u001b[0m -  Output directory /Users/jhc/_Devel/cookbook/recipes/manual_alg\n",
      "\u001b[32mINFO\u001b[0m -  Erasing all existing npr data...\n",
      "\u001b[1;37;43mWRNG\u001b[0m -  Using existing dir: /Users/jhc/_Devel/cookbook/recipes/manual_alg/db\n",
      "\u001b[32mINFO\u001b[0m -  Reading aa sequences from data/P53.alg.fa...\n",
      "\u001b[1;37;43mWRNG\u001b[0m -  8 target sequences\n",
      "\u001b[32mINFO\u001b[0m -  ETE build starts now!\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:42 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[34mMultiSeqTask\u001b[0m (8 aa seqs, MSF, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[34mMultiSeqTask\u001b[0m (8 aa seqs, MSF, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[1;37;43mWRNG\u001b[0m -          Skipping alignment phase, using original sequences\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:44 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[31mAlgTask\u001b[0m (8 aa seqs, ManualAlg, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[31mAlgTask\u001b[0m (8 aa seqs, ManualAlg, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:46 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[35mModelTesterTask\u001b[0m (8 aa seqs, PModelTest-[JTT,WAG,VT,LG,MtREV], /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:48 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[35mModelTesterTask\u001b[0m (8 aa seqs, PModelTest-[JTT,WAG,VT,LG,MtREV], /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:50 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;31mR\u001b[0m) \u001b[35mModelTesterTask\u001b[0m (8 aa seqs, PModelTest-[JTT,WAG,VT,LG,MtREV], /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[35mModelTesterTask\u001b[0m (8 aa seqs, PModelTest-[JTT,WAG,VT,LG,MtREV], /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:52 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[32mTreeTask\u001b[0m (8 aa seqs, RaxML, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:54 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[32mTreeTask\u001b[0m (8 aa seqs, RaxML, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:56 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[32mTreeTask\u001b[0m (8 aa seqs, RaxML, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:17:58 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[32mTreeTask\u001b[0m (8 aa seqs, RaxML, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:18:00 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (Q\u001b[0m) \u001b[32mTreeTask\u001b[0m (8 aa seqs, RaxML, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[32mTreeTask\u001b[0m (8 aa seqs, RaxML, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40m Updating tasks status:\u001b[0m (Mon Feb  8 14:18:02 2016)\n",
      "\u001b[32mINFO\u001b[0m -  Thread \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m: pending tasks: \u001b[1;33m1\u001b[0m of sizes: 8\n",
      "\u001b[32mINFO\u001b[0m -   (W\u001b[0m) \u001b[36mTreeMergeTask\u001b[0m (8 aa seqs, TreeMerger, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -   (\u001b[1;34mD\u001b[0m) \u001b[36mTreeMergeTask\u001b[0m (8 aa seqs, TreeMerger, /\u001b[1;37;40mnone-none-...ml_default\u001b[0m)\n",
      "\u001b[32mINFO\u001b[0m -  \u001b[1;37;40mWaiting 2 seconds\u001b[0m\n",
      "\u001b[32mINFO\u001b[0m -  Assembling final tree...\n",
      "\u001b[32mINFO\u001b[0m -  Done thread \u001b[1;37;45mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m in 1 iteration(s)\n",
      "\u001b[32mINFO\u001b[0m -  Writing final tree for \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m\n",
      "   /Users/jhc/_Devel/cookbook/recipes/manual_alg/none-none-pmodeltest_soft_ultrafast-raxml_default/P53.alg.fa.final_tree.nw\n",
      "   /Users/jhc/_Devel/cookbook/recipes/manual_alg/none-none-pmodeltest_soft_ultrafast-raxml_default/P53.alg.fa.final_tree.nwx (newick extended)\n",
      "\u001b[32mINFO\u001b[0m -  Writing final tree alignment \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m\n",
      "   /Users/jhc/_Devel/cookbook/recipes/manual_alg/none-none-pmodeltest_soft_ultrafast-raxml_default/P53.alg.fa.final_tree.used_alg.fa\n",
      "\u001b[32mINFO\u001b[0m -  Writing root node alignment \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m\n",
      "   /Users/jhc/_Devel/cookbook/recipes/manual_alg/none-none-pmodeltest_soft_ultrafast-raxml_default/P53.alg.fa.final_tree.fa\n",
      "\u001b[32mINFO\u001b[0m -  Generating tree image for \u001b[1;37;40mnone-none-pmodeltest_soft_ultrafast-raxml_default\u001b[0m\n",
      "   /Users/jhc/_Devel/cookbook/recipes/manual_alg/none-none-pmodeltest_soft_ultrafast-raxml_default/P53.alg.fa.final_tree.png\n",
      "\u001b[32mINFO\u001b[0m -  Done\n",
      "\u001b[32mINFO\u001b[0m -  Deleting temporal data...\n",
      "   ========================================================================\n",
      "         The following published software and/or methods were used.        \n",
      "               *** Please, do not forget to cite them! ***                 \n",
      "   ========================================================================\n",
      "   Guindon S, Dufayard JF, Lefort V, Anisimova M, Hordijk W, Gascuel O. New\n",
      "      algorithms and methods to estimate maximum-likelihood phylogenies:\n",
      "      assessing the performance of PhyML 3.0. Syst Biol. 2010\n",
      "      May;59(3):307-21.\n",
      "   Huerta-Cepas J, Dopazo J, Gabaldón T. ETE: a python Environment for Tree\n",
      "      Exploration. BMC Bioinformatics. 2010 Jan 13;11:24.\n",
      "   Stamatakis A. RAxML version 8: a tool for phylogenetic analysis and\n",
      "      post-analysis of large phylogenies Bioinformatics (2014) 30 (9):\n",
      "      1312-1313.\n"
     ]
    }
   ],
   "source": [
    "%%bash \n",
    "ete3 build -a data/P53.alg.fa -w none-none-pmodeltest_soft_ultrafast-raxml_default -o manual_alg --clearall "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
