
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="ete-build">ete-build<a class="anchor-link" href="#ete-build">&#182;</a></h1><p>Build phylogenetic trees using reproducible workflows and executed a single command line</p>
<p><em>ete-build</em> is a wrapper tool that simplifies the reconstruction of <strong>gene trees</strong> and <strong>species trees</strong> by executing custom workflows. Most notable features include:</p>
<ul>
<li><p>All the external software required to align sequences, test evolutionary models and infer trees is handled transparently and pipelined as necessary.</p>
</li>
<li><p>Full gene-tree and species-tree workflows can be executed with a single command line.</p>
</li>
<li><p>Choose between simple or advance use:</p>
<ul>
<li>A number of pre-configured workflows are provided. You can just pick a name and run it</li>
<li>Custom workflows can be build using a gallery of pre-configured application binding</li>
<li>New workflows and application binds can be created</li>
</ul>
</li>
<li><p>Support for multi-CPU systems. Tasks are distributed in parallel taking into account data dependencies and enabling multi-threading options in external software when avaialble.</p>
</li>
<li><p>HPC ready: As workflows run with a single self contained process, computing thousands of trees using in a Cluster is as easy as submiting one job per tree.</p>
</li>
<li><p>User-friendly monitoring: All tasks are monitored during execution.</p>
</li>
<li><p>Extendable. Unsupported software can be plugged-in by developing simple bindings.</p>
</li>
</ul>
<h1 id="Getting-started">Getting started<a class="anchor-link" href="#Getting-started">&#182;</a></h1><h2 id="Build-a-gene-phylogeny">Build a gene phylogeny<a class="anchor-link" href="#Build-a-gene-phylogeny">&#182;</a></h2><p>The easiest way to build gene-tree is to <a href="http://etetoolkit.org/ete-build">choose</a> a name from the list of predefined workflows and run it.</p>
<h2 id="Build-a-species-phylogeny">Build a species phylogeny<a class="anchor-link" href="#Build-a-species-phylogeny">&#182;</a></h2><p><code>ete-builds</code> allows bulding species trees based on concatenated alignments. All you need to provide is sequences and the set of gene families to concatenate (Usually referred as Clusters of Orthologous Groups - COGs)</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered">
<div class="prompt input_prompt">
</div>
<div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Documentation">Documentation<a class="anchor-link" href="#Documentation">&#182;</a></h1><ul>
<li><a href="ete_build_basics.ipynb">Basic concepts behind ete-build</a> </li>
<li><a href="ete_build_workflows.ipynb">Personalize workflows</a> </li>
<li><p><a href="ete_build_custom_workflows.ipynb">Create new workflows and application bindings</a></p>
</li>
<li><p><a href="ete_build_mixed_types.ipynb">Combining amino-acid and nucleotide sequences</a></p>
</li>
<li><a href="ete_build_resuming.ipynb">Resuming a broken analysis</a></li>
<li><p><a href="ete_build_metaworkflows.ipynb">Runninng multiple workflows at once</a></p>
</li>
<li><p><a href="ete_build_supematrix.ipynb">Building species trees</a></p>
</li>
<li><a href="ete_build_input_seqs.ipynb">Tips and options regarding input data</a></li>
<li><a href="ete_build_cluster.ipynb">Tips for running in a cluster environment</a></li>
<li><a href="ete_build_troubleshoot.ipynb">Troubleshooting</a></li>
</ul>
<h1 id="About-reproducibility">About reproducibility<a class="anchor-link" href="#About-reproducibility">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-ipython2"><pre> 
</pre></div>

</div>
</div>
</div>

</div>